### PR TITLE
removing `out T` from docs since it no longer working

### PR DESCRIPTION
**Pull request by codehz**
see https://github.com/nim-lang/Nim/issues/16131

### CODE REVIEW

File: doc/manual.rst [hunk 1]
```diff
@@ -2502,10 +2502,10 @@ matches) is preferred:
 Overloading based on 'var T' / 'out T'
```
Comment (timotheecour) [inline doc/manual.rst]: missed this: `Overloading based on 'var T' / 'out T'`

**Comment by timotheecour** [comment]
/cc @zah since you introduced these in 7e0c66ffe79f1616c1c821ede5f50f03406feabe and 5e368f36390a8164abbedf2f6c79ead0f510df29

`out` is also referred to in `doc/manual_experimental.rst` eg this snippet:

```nim
  type
    AnnotatedPtr[out T] =
      metadata: MyTypeInfo
      p: ref T

    RingBuffer[out T] =
      startPos: int
      data: seq[T]

    Action {.importcpp: "std::function<void ('0)>".} [in T] = object
```

(which is really wrong, eg misses `object` after `=`, reason more to use runnableExamples everywhere)
but after partially fixing it, I get:

```nim
type
  AnnotatedPtr[out T] = object
    metadata: MyTypeInfo
    p: ref T
  RingBuffer[out T] = object
    startPos: int
    data: seq[T]
  Action {.importcpp: "std::function<void ('0)>".} [in T] = object
```

```
Error: the 'out' modifier can be used only with imported types
    AnnotatedPtr[out T] = object
                 ^
/Users/timothee/git_clone/nim/Nim_devel/compiler/semtypes.nim(2124, 25) compiler msg initiated here [MsgOrigin]
```

I don't understand this feature as there is no working example and 5e368f36390a8164abbedf2f6c79ead0f510df29 introduced the feature but only a failing test, `tests/errmsgs/tinvalidinout.nim`, not a working test. What is an example of a 'out' modifier with an imported type?

**Review by timotheecour — APPROVED** [review]
LGTM; https://github.com/nim-lang/Nim/pull/16378#issuecomment-747606595 can be dealt with later

File: doc/manual.rst [hunk 2]
```diff
@@ -2524,7 +2524,7 @@ type checking, the argument is checked to be an `l-value`:idx:.
               # 13
 
 
-An l-value matches ``var T`` and ``out T`` equally well, hence
+An l-value matches ``var T`` equally well, hence
```
Comment (Araq) [inline doc/manual.rst]: The full paragraph needs to be removed.

**Pull request merged**